app/src/main/java/fr/cedriccreusot/pokedex/PokemonModule.kt

```diff
@@ -0,0 +1,27 @@
+package fr.cedriccreusot.pokedex
+
+import PokemonRepositoryAdapter
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ActivityRetainedComponent
+import fr.cedriccreusot.domain.list.repository.PokemonRepository
+import fr.cedriccreusot.domain.list.usecase.FetchPokemonListUseCase
+import me.sargunvohra.lib.pokekotlin.client.PokeApi
+import me.sargunvohra.lib.pokekotlin.client.PokeApiClient
+
+@Module
+@InstallIn(ActivityRetainedComponent::class)
+class PokemonModule {
+    @Provides
+    fun providePokeApi(): PokeApi = PokeApiClient()
+
+    @Provides
+    fun providePokemonRepository(
+        pokemonDataSource: PokeApi
+    ): PokemonRepository = PokemonRepositoryAdapter(pokemonDataSource)
```

Comment:
If I follow the documentation you can use @Binds to give the interface to use.

Reply:
True